Add support for Azure

arturowczarek opened the pull request:
* The image and the Helm chart can read the events logs from the Azure Data Lake Storage Gen2
* The Helm chart creates and uses the secret to pull image from ghcr.io
* The script to launch the server locally, can read the configurations from .env files

vara-bonthu commented:
@arturowczarek Thanks for extending the Spark History Server to support Azure!
Looking forward to the finalized PR. We'll need to run tests to verify the solution works with both AWS and Azure
environments.

arturowczarek commented:
@vara-bonthu 
I've introduced all the necessary changes:

- I've restructured the values.yaml file:
    - I've introduced a convenient way to create image pull credentials for the ones who don't know how to create them, but can generate a GitHub token to use the GitHub Containers Repository. To use it, you have to enable image.pullCredentials.enabled and provide the properties
    - I've moved imagePullSecrets to image section so that it is close to pullCredentials. You can use either of them
- I've created a section logStore where you switch between s3 and abfs mode. When you pick your mode, you have to provide the properties
- All the values.yaml properties are governed by appropriate json schema. So you have to pick either s3 or abfs mode. Otherwise Helm complains. If you pick the mode, the schema makes sure all the placeholders are replaced with the actual values. The same validation is for image.pullCredentials
- I've moved all the settings for history server to config map
- I've added a bunch of the most popular history server settings there as well
- I've added the empty writable volume for history server logs, because the application was complaining
- I've created Helm unit tests to make sure everything works as intended. It speeds up the development
- I've added a GitHub workflow to run the Helm unit tests on pull request. If someone does something wrong, the rendered templates are uploaded and they can download them and see what is wrong
- In Docker image:
  - I've updated the pom.xml to include also hadoop-azure dependencies and separated them as a Maven profile
  - I've updated the script to launch the image locally
  - The script can read the values from .env file if it is available. The file is ignored by default so that someone doesn't commit it by mistake.
- I've updated the readme. It can still be improved, but I have to stop the work somewhere

I've tested the everything with Azure (I had released the image in my own repository and deployed it).

I'd need help to test it with AWS.

We can make a call if you need to ask questions and see it in action.

vara-bonthu commented on docker/Dockerfile:
```diff
@@ -20,15 +20,15 @@ COPY pom.xml /tmp
 # into Spark's JAR directory. This ensures S3/Hadoop/AWS SDK integration.
 # Then remove conflicting or outdated jars to prevent runtime classloader issues.
-RUN mvn dependency:copy-dependencies -DoutputDirectory=/opt/spark/jars/ && \
+RUN mvn -Paws -Pazure dependency:copy-dependencies -DoutputDirectory=/opt/spark/jars/ && \
```
add this at the top of the file `ARG MAVEN_PROFILE=aws`. 
Github Docker publish can be modified to generate images for both aws and azure like this 

```
ghcr.io/kubedai/spark-history-server:aws-1.3.2
ghcr.io/kubedai/spark-history-server:azure-1.3.2
ghcr.io/kubedai/spark-history-server:aws-latest
ghcr.io/kubedai/spark-history-server:azure-latest
```

vara-bonthu commented:
Will take a look at it today and merge it accordingly.
